sha2_256.c

```diff
@@ -143,6 +143,13 @@ void sha2_256_compress(void *v)
 /* initialize */
 
 static void sha2_256_init_h0(sha2_256_t *sha, const uint8_t *h0)
+__contract__(
+  requires(memory_no_alias(sha, sizeof(sha2_256_t)))
+  requires(memory_no_alias(h0, 32))
+  assigns(memory_slice(sha->s, sizeof(uint32_t)*(8+24)))
+  assigns(sha->i)
+  assigns(sha->len)
```

Comment:
It seems that all fields of `sha` are assigned to, right?
Maybe we should just write:
```
assigns(object_whole(sha))
```
?